crates/node/src/commands/run.rs

```diff
@@ -2,15 +2,48 @@ use std::path::PathBuf;
 
 use madara_runtime::SealingMode;
 use mc_data_availability::DaLayer;
-use mc_l1_messages::config::L1MessagesWorkerConfig;
-use mc_l1_messages::error::L1MessagesConfigError;
+use mc_l1_messages::config::{L1MessagesWorkerConfig, L1MessagesWorkerConfigError};
 use sc_cli::{Result, RpcMethods, RunCmd, SubstrateCli};
 use sc_service::BasePath;
 use serde::{Deserialize, Serialize};
 
 use crate::cli::Cli;
 use crate::service;
 
+#[derive(Debug, Clone, clap::Args)]
+pub struct L1MessagesParams {
+    /// Ethereum Provider (Node) Url
+    #[clap(
+        long,
+        requires="l1_contract_address",
+        value_hint=clap::ValueHint::Url,
+    )]
+    pub provider_url: String,
+
+    /// L1 Contract Address
+    #[clap(
+        long,
+        requires="provider_url",
+        value_hint=clap::ValueHint::Other,
+    )]
+    pub l1_contract_address: String,
+}
+
+#[derive(Debug, Clone, clap::Args)]
+pub struct L1Messages {
+    /// Path to configuration file for Ethereum Core Contract Events Listener
+    #[clap(
+        long,
+        conflicts_with="provider_url",
+        conflicts_with="l1_contract_address",
```

Comment:
I think you can make it conflict with the ArgGroup directly

Reply:
```
thread 'main' panicked at /Users/danielbigos/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.4.8/src/builder/debug_asserts.rs:214:13:
Command madara: Argument or group 'config_params' specified in 'conflicts_with*' for 'l1_messages_config' does not exist
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Cannot use `exclusive=true`
```sh
Running `target/debug/madara --dev --l1-messages-config ./examples/messaging/l1-messaging-config.json`
error: the argument '--l1-messages-config <L1_MESSAGES_CONFIG>' cannot be used with one or more of the other specified arguments

Usage: madara [OPTIONS] --provider-url <PROVIDER_URL> --l1-contract-address <L1_CONTRACT_ADDRESS>
       madara <COMMAND>
```